File: sic_home/lib/models/create_device_model.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:sic_home/models/device.dart';

part 'create_device_model.g.dart';

@JsonSerializable()
class CreateDeviceModel {
  String name;
  DeviceType deviceType;
  String userId;
  int placeId;
  int sectionId;
  int pin;

  CreateDeviceModel({
    required this.name,
    required this.deviceType,
    required this.userId,
    required this.placeId,
    required this.sectionId,
    required this.pin,
  });

  factory CreateDeviceModel.fromJson(Map<String, dynamic> json) =>
      _$CreateDeviceModelFromJson(json);

  Map<String, dynamic> toJson() => _$CreateDeviceModelToJson(this);
}

File: SIC_Backend/Data/Models/CreateDeviceModel.cs
namespace SIC_Backend.Data.Models
{
    public class CreateDeviceModel
    {
        public required string Name { get; set; }
        public required DeviceTypes DeviceType { get; set; }
        public required string UserId { get; set; }
        public required int PlaceId { get; set; }
        public required int SectionId { get; set; }
        public required int Pin { get; set; }
    }
}